core/src/main/java/jenkins/model/queue/ItemDeletion.java

```diff
@@ -266,12 +266,10 @@ public static void cancelBuildsInProgress(@NonNull Item initiatingItem) throws F
                     // comparison with executor.getCurrentExecutable() == executable currently should always be
                     // true as we no longer recycle Executors, but safer to future-proof in case we ever
                     // revisit recycling.
-                    if (!entry.getKey().isAlive()
+                    if (!entry.getKey().isActive()
                             || entry.getValue() != entry.getKey().getCurrentExecutable()) {
                         iterator.remove();
                     }
-                    // I don't know why, but we have to keep interrupting
-                    entry.getKey().interrupt(Result.ABORTED);
```

Comment:
I removed this after some discussion in https://github.com/jenkinsci/workflow-job-plugin/pull/468#discussion_r1777740255. The fact that we were repeatedly interrupting things here was a bit unusual. The extra interruptions were added in https://github.com/jenkinsci/jenkins/commit/047e849d09a914ec0178d98e6f6be681e0b213e5, but the test in that commit passes even without the extra interruptions. For Pipelines, which are the only jobs I know of where `interrupt` may not kill the job promptly, repeatedly calling `interrupt` is not going to help, you have to move to more severe methods such as `doTerm` and `doKill` (perhaps we should consider making Pipeline do this on its own). Now that Pipelines are handled correctly, the repeated interruptions every 50ms could produce a large number of timer tasks [here](https://github.com/jenkinsci/workflow-job-plugin/blob/d82e9f82c8115f04fa3f27fc38e18ca01c6410d2/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L385-L403) for Pipelines that take a while to complete, whereas previously a single Pipeline would have been interrupted at most twice.

Reply:
For Pipelines, the main issue is that users are very easily able to inadvertently swallow interruptions just by adding a `try`/`catch` block to their Pipeline script. [`FlowInterruptedException.actualInterruption`](https://github.com/jenkinsci/workflow-step-api-plugin/blob/f6d31e6b744fa6a83d58d980d027ea7fe788dc16/src/main/java/org/jenkinsci/plugins/workflow/steps/FlowInterruptedException.java#L60-L67) makes it possible to avoid swallowing exceptions in trusted scripts and from Pipeline steps, but I would guess there are many steps that do not handle this correctly, and sandboxed scripts do not have access to this field by default.

For non-Pipelines, poorly written `Builder` and `Publisher` implementations may also swallow interruptions, but I have no idea how common this is.

What do other people think about this? I see 3 main options:
* Interrupt builds only once (as in this PR now) - if the build doesn't respond to interruption, someone will have to kill it some other way or retry deletion until the build finally dies. This is how interruption works if you try to cancel a build in the Jenkins UI.
* Interrupt builds repeatedly in a loop every 50ms until they die (as before this PR) - Best chance of interrupting the build, but is not aligned with other interruption mechanisms in Jenkins, and is still not guaranteed to cancel builds in all cases. In combination with the main fix here, this can lead to a large number of timer tasks being created when interrupting Pipeline builds, which seems undesirable. For example, if a build takes the full 15s to die, at 50ms per interrupt we will try to interrupt the build 300 times.
* Interrupt builds repeatedly, but at a much lower frequency, say once per second. This would make this code  able to cancel builds that swallow a single interruption due to something simple like a `try`/`catch` block in a Pipeline, but would avoid overloading the Pipeline interruption mechanism.

Reply:
First option looks fine to me. Users will try to interrupt if they see nothing happens.

Reply:
The first option seems appropriate for purposes of this PR; probably `WorkflowRun` (and/or `CpsFlowExecution`) should be independently improved to enforce an exit from an `actualInterruption` in a timely manner.